File: Leetcode/PY/Leetcode475-供暖器.py
class Solution:
    def findRadius(self, houses: List[int], heaters: List[int]) -> int:
        l,r = 0,int(1e9)
        houses.sort()
        heaters.sort()
        def check(length):
            i,j = 0,0
            while (i < len(houses)):
                while j < len(heaters) and abs(houses[i] - heaters[j]) > length:
                    j += 1
                if j >= len(heaters):
                    return False
                i += 1
            return True
        while l < r:
            mid = l + r >> 1
            if(check(mid)): r = mid
            else: l = mid + 1
        return r

File: Leetcode/JS/Leetcode475-供暖器.js
var findRadius = function(houses, heaters) {
    houses.sort((a,b)=> a - b);
    heaters.sort((a,b) => a - b);
    let l = 0,r = 1e9;
    let check = function(len){
        for(let i = 0,j = 0;i < houses.length;i++){
            while(j < heaters.length && Math.abs(houses[i] - heaters[j]) > len){
                j++; 
            } 
            if(j >= heaters.length) return false;
        }
        return true;
    }
    while(l < r){
        let mid = (l + r) >> 1;
        if(check(mid)) r = mid;
        else l = mid + 1;
    }
    return r;
};